src/main/scala/com.mediative.eigenflow/domain/RecoveryStrategy.scala

```diff
@@ -18,19 +18,16 @@ package com.mediative.eigenflow.domain
 
 import scala.concurrent.duration.FiniteDuration
 
-/**
- * Retry strategy.
- *
- * @param interval Interval between retries.
- * @param attempts Number of retries.
- */
-case class Retry(interval: FiniteDuration, attempts: Int)
+trait RecoveryStrategy
+
+object RecoveryStrategy {
+  case class Retry(interval: FiniteDuration, attempts: Int) extends RecoveryStrategy
 
-object Retry {
+  case object Fail extends RecoveryStrategy
 
-  import scala.concurrent.duration._
+  case object Complete extends RecoveryStrategy
```

Comment:
Let's start with that.  We can always add more later if we decide it's not informative enough.